src/encoding/monoid.rs

```diff
@@ -0,0 +1,23 @@
+// Copyright 2022 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+// Also licensed under MIT license, at your choice.
+
+/// Interface for a monoid.
```

Comment:
Probably good to mention that default is the identity of the monoid.

I'm also curious why T is a generic parameter rather than an associated type.

The existence of the `new` method makes this a monoid homomorphism rather than a vanilla monoid, but for the purposes of this module we don't have to go full math nerd.

Reply:
Associated type seems reasonable to me. Also, always happy to be educated in category theory :)